gsy_e_sdk/setups/Asset_API_scripts/redis_settle_energy_deviations.py

```diff
@@ -0,0 +1,181 @@
+"""
+Template file to post energy deviations as bids
+and offers in the Settlement Market with Redis.
+"""
+
+from time import sleep
+from typing import List, Dict
+from pendulum import from_format
+from gsy_framework.constants_limits import DATE_TIME_FORMAT
+from gsy_e_sdk.redis_aggregator import RedisAggregator
+from gsy_e_sdk.clients.redis_asset_client import RedisAssetClient
+
+ORACLE_NAME = "oracle"
+
+# List of assets' names to be connected with the API
+LOAD_NAMES = ["H1 General Load", "H2 General Load"]
+PV_NAMES = ["H1 PV", "H2 PV"]
+STORAGE_NAMES = []
+
+
+class Oracle(RedisAggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.send_forecasts(market_info)
+        self.send_measurements(market_info)
+        self.settle_energy_deviations()
+        self.post_bid_offer()
+        self.execute_batch_commands()
+
+    def send_forecasts(self, market_info):
+        """Send forecasts of the next market slot to the exchange."""
+        forecast_market_slot = (
+            from_format(market_info["market_slot"], DATE_TIME_FORMAT)
+            .add(minutes=15)
+            .format(DATE_TIME_FORMAT)
+        )
+        # pylint: disable=unused-variable
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if "asset_info" not in area_dict or area_dict["asset_info"] is None:
+                continue
+            # Consumption assets
+            if "energy_requirement_kWh" in area_dict["asset_info"]:
+
+                self.add_to_batch_commands.set_energy_forecast(
+                    asset_uuid=area_uuid,
+                    energy_forecast_kWh={forecast_market_slot: 1.2},
```

Comment:
I understand your concerns and I agree with you, I am just not sure how write it efficiently and clearly generically once in the whole PR. Anyways, this will be specified in the wiki.